### PR TITLE
Fixed HeFFTe package spec to not do the smoke test prior to 2.2.0

### PR DESCRIPTION
**Pull request by patrickb314**
HeFFTe smoke test setup code breaks spack installs with HeFFTe prior to 2.2.0 (specifically, for example, building Cabana which currently requires 2.1.0). This pull request disables the smoke test code prior to version 2.2.0.

### CODE REVIEW

**Comment by mkstoyanov** [comment]
I see, the spack testing was rocky in the old version and didn't think anyone will need it. I'm OK with the change.

**Review by mkstoyanov — APPROVED** [review]

**Comment by G-Ragghianti** [comment]
I would like to test this.  In what way does this break the install of other packages?  Can we have a printout that shows that the tests were skipped?

**Comment by patrickb314** [comment]
From my mac with homebrew/spack (but seeing this on a WSL system running Ubuntu, the UNM HPC systems, and the LLNL Lassen):
> bridges@patricks-mbp-2[111]> spack install heffte@2.1.0 +fftw
> ==> Installing heffte-2.1.0-pampzfkkddwc2qou23bhbhzsh5kisuft [25/25]
> ==> No binary for heffte-2.1.0-pampzfkkddwc2qou23bhbhzsh5kisuft found: installing from source
> ==> Using cached archive: /Users/bridges/spack/var/spack/cache/_source-cache/archive/63/63b8ea45a220afc4fa0b14769c0dd291e614a2fe9d5a91c50d28f16ee29b3f1c.tar.gz
> ==> No patches needed for heffte
> ==> heffte: Executing phase: 'cmake'
> ==> heffte: Executing phase: 'build'
> ==> heffte: Executing phase: 'install'
> ==> Error: OSError: No such file or directory: '/Users/bridges/spack/opt/spack/darwin-ventura-m2/apple-clang-14.0.3/heffte-2.1.0-pampzfkkddwc2qou23bhbhzsh5kisuft/share/heffte/testing'
> 
> /Users/bridges/spack/var/spack/repos/builtin/packages/heffte/package.py:131, in setup_smoke_test:
>         130    def setup_smoke_test(self):
>   >>    131        install_tree(
>         132            self.prefix.share.heffte.testing, join_path(self.install_test_root, "testing")
>         133        )
> 
> See build log for details:
>   /var/folders/yg/2794s5sx4h3cjj3rtpmtyvy80000gn/T/bridges/spack-stage/spack-stage-heffte-2.1.0-pampzfkkddwc2qou23bhbhzsh5kisuft/spack-build-out.txt
> 
> bridges@patricks-mbp-2[112]>

**Comment by patrickb314** [comment]
With the patch: 

> bridges@dhcp102[31]> spack install heffte@2.1.0 +fftw
> ...
> ==> Installing heffte-2.1.0-ihmow5edj3gpj3cuntsbpw7g5xzulj2e [25/25]
> ==> No binary for heffte-2.1.0-ihmow5edj3gpj3cuntsbpw7g5xzulj2e found: installing from source
> ==> Using cached archive: /Users/bridges/spack/var/spack/cache/_source-cache/archive/63/63b8ea45a220afc4fa0b14769c0dd291e614a2fe9d5a91c50d28f16ee29b3f1c.tar.gz
> ==> No patches needed for heffte
> ==> heffte: Executing phase: 'cmake'
> ==> heffte: Executing phase: 'build'
> ==> heffte: Executing phase: 'install'
> ==> heffte: Successfully installed heffte-2.1.0-ihmow5edj3gpj3cuntsbpw7g5xzulj2e
>   Stage: 0.01s.  Cmake: 0.54s.  Build: 5.45s.  Install: 0.18s.  Post-install: 0.04s.  Total: 6.31s
> [+] /Users/bridges/spack/opt/spack/darwin-ventura-m2/apple-clang-14.0.3/heffte-2.1.0-ihmow5edj3gpj3cuntsbpw7g5xzulj2e

**Comment by mkstoyanov** [comment]
The files that the spack smoke test is using were added in v2.2.0.

There is no spack smoke test in v2.1.0 or earlier. There is a smoke test via a bash script, but that creates a ton of conflicts with spack, using that script was not sustainable and that's why the smoke test was changed.

**Comment by patrickb314** [comment]
A little more background. We're preparing our Beatnik fluid interface solver for it's first release (https://github.com/CUP-ECS/Beatnik), and it relies on the ECP-COPA Cabana/Cajita framework, which currently relies on HeFFTe 2.1.0. Cabana 0.6 will be released soon and will roll forward to a later HeFFTe release, but this PR will enable any code that depend on earlier versions of HeFFTe to install it using spack.

**Comment by patrickb314** [comment]
@G-Ragghianti Does that address your concerns?

File: var/spack/repos/builtin/packages/heffte/package.py [hunk 1]
```diff
@@ -128,12 +128,18 @@ def cmake_args(self):
...
+            return
         install_tree(
             self.prefix.share.heffte.testing, join_path(self.install_test_root, "testing")
         )
 
     def test_make_test(self):
         """build and run make(test)"""
+
+        if self.spec.satisfies("@:2.2.0"):
+            return
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/heffte/package.py]: The docs are probably still lacking on this feature.

I suggest raising the `SkipTest` exception so the test cab be flagged `SKIPPED` and picked up by CI/CDash.

```suggestion
            raise SkipTest("Test is not supported for versions @:2.2.0")
```

**Review by tldahlgren — COMMENTED** [review]
LGTM.  Just suggest a change so CDash/CI will flag the test as skipped.

**Comment by G-Ragghianti** [comment]
Yes, looks good. Thanks for the patch.

**Review by G-Ragghianti — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**